core/src/test/java/feast/core/logging/CoreLoggingIT.java

```diff
@@ -0,0 +1,201 @@
+      throws InterruptedException, InvalidProtocolBufferException {
+    // Generate artifical load on feast core.
+    UpdateStoreRequest request =
+        UpdateStoreRequest.newBuilder().setStore(DataGenerator.getDefaultStore()).build();
+    UpdateStoreResponse response = coreService.updateStore(request);
+
+    // Wait required to ensure audit logs are flushed into test audit log appender
+    Thread.sleep(1000);
+    // Check message audit logs are produced for each audit log.
+    JsonFormat.Parser protoJSONParser = JsonFormat.parser();
+    // Pull message audit logs logs from test log appender
+    List<JsonObject> logJsonObjects =
+        parseMessageJsonLogObjects(testAuditLogAppender.getLogs(), "UpdateStore");
+    assertEquals(1, logJsonObjects.size());
+    JsonObject logObj = logJsonObjects.get(0);
+
+    // Extract & Check that request/response are returned correctly
+    String requestJson = logObj.getAsJsonObject("request").toString();
+    UpdateStoreRequest.Builder gotRequest = UpdateStoreRequest.newBuilder();
+    protoJSONParser.merge(requestJson, gotRequest);
+
+    String responseJson = logObj.getAsJsonObject("response").toString();
+    UpdateStoreResponse.Builder gotResponse = UpdateStoreResponse.newBuilder();
+    protoJSONParser.merge(responseJson, gotResponse);
+
+    assertThat(gotRequest.build(), equalTo(request));
+    assertThat(gotResponse.build(), equalTo(response));
+  }
+
+  /** Check that message audit logs are produced when server encounters an error */
+  @Test
+  public void shouldProduceMessageAuditLogsOnError() throws InterruptedException {
+    // Send a bad request which should cause Core to error
+    ListFeatureSetsRequest request =
+        ListFeatureSetsRequest.newBuilder()
+            .setFilter(
+                ListFeatureSetsRequest.Filter.newBuilder()
+                    .setProject("*")
+                    .setFeatureSetName("nop")
+                    .build())
+            .build();
+
+    boolean hasExpectedException = false;
+    Code statusCode = null;
+    try {
+      coreService.listFeatureSets(request);
+    } catch (StatusRuntimeException e) {
+      hasExpectedException = true;
+      statusCode = e.getStatus().getCode();
+    }
+    assertTrue(hasExpectedException);
+
+    // Wait required to ensure audit logs are flushed into test audit log appender
+    Thread.sleep(1000);
+    // Pull message audit logs logs from test log appender
+    List<JsonObject> logJsonObjects =
+        parseMessageJsonLogObjects(testAuditLogAppender.getLogs(), "ListFeatureSets");
+
+    assertEquals(1, logJsonObjects.size());
+    JsonObject logJsonObject = logJsonObjects.get(0);
+    // Check correct status code is tracked on error.
+    assertEquals(logJsonObject.get("statusCode").getAsString(), statusCode.toString());
+  }
+
+  /** Check that expected message audit logs are produced when under load. */
+  @Test
+  public void shouldProduceExpectedNumberOfAuditLogs()
```

Comment:
I thought the point of the race was that it would always log something but would just have some missing data. Is the test you are doing here only to make sure something was logged? Seems like a good idea to make sure the contents itself is correct.

Reply:
Updated the test to check  the request and response are correct.